xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -302,18 +302,21 @@ enum AppAction: Hashable {
     case mergeEntry(parent: Slashlink, child: Slashlink)
     case moveEntry(from: Slashlink, to: Slashlink)
     case updateAudience(address: Slashlink, audience: Audience)
+    case assignColor(addess: Slashlink, color: NoteColor)
```

Comment:
I think I would recommend just passing `Color` at the level of actions. `NoteColor` can act as a constant we use when assigning colors.

Rationale: the plumbing then supports any color. We enforce defaults at the io boundary. This would allow users to hack the color headers, for example.

If the intent is to support dynamically adaptive colors for light/dark, I believe SwiftUI supports this through the asset bundle (specify light/dark variants of color constants).

Alternatively, if we wanted to keep this in code, we might create a more general type like `AdaptiveColor`:

```swift
struct AdaptiveColor: Hashable {
    var light: Color
    var dark: Color

    func color(for scheme: ColorScheme) -> Color {
        switch scheme {
        case .light:
            return light
        case .dark:
            return dark
        }
    } 
}
```

Reply:
I moved the colors into the asset bundle, see response here https://github.com/subconsciousnetwork/subconscious/pull/1064#issuecomment-1884141408